### PR TITLE
Refactoring NimBLEAddress

### PR DESCRIPTION
**Pull request by Jeroen88**
- Add logging for malformed addresses
- Set invalid addresses to "00:00:00:00:00:00", so the main program can easily check if an address is correctly parsed (`if(NimBLEAddress("invalid address") == NimBLEAddress(0)) ...`)
- Add constructor directly from a (hex) uint64_t value. `NimBLEAddress(0xa4c1385def16)` yields the same address as `NimBLEAddress("a4:c1:38:5d:ef:16")`
- Remove `memrcpy()`, using standard cpp instead
- Add `operator ==`, `operator !=` and `operator std::string()` cast
- Use `operator ==` in `::equals()`
- Use `operator std::string()` in `::toString()` (also avoiding the `malloc` because the cast uses a local buffer that is declared on the stack)

### CODE REVIEW

File: src/NimBLEAddress.cpp [hunk 1]
```diff
@@ -44,25 +48,39 @@ NimBLEAddress::NimBLEAddress(ble_addr_t address) {
...
 
 
 /**
- * @brief Constructor for compatibility with bluedrioid esp library.
+ * @brief Constructor for compatibility with bluedroid esp library.
  * @param [in] uint8_t[6] or esp_bd_addr_t struct containing the address.
  */
 NimBLEAddress::NimBLEAddress(uint8_t address[6]) {
-    NimBLEUtils::memrcpy(m_address, address, 6);
+    memcpy(m_address, address, sizeof m_address);
```
Comment (h2zero) [inline src/NimBLEAddress.cpp]: Any reason to not just use `std::reverse_copy` here?

File: src/NimBLEAddress.cpp [hunk 2]
```diff
@@ -44,25 +48,39 @@ NimBLEAddress::NimBLEAddress(ble_addr_t address) {
...
+    }
+    for(size_t index = 0; index < sizeof m_address; index++) {
+        m_address[index] = data[index];
+    }
 } // BLEAddress
 
 
 /**
- * @brief Constructor for compatibility with bluedrioid esp library.
+ * @brief Constructor for compatibility with bluedroid esp library.
```
Comment (h2zero) [inline src/NimBLEAddress.cpp:69]: lol, good catch!

**Comment by h2zero** [comment]
Minor comment, but looks good! I'll give it some testing, thanks 😄.

File: src/NimBLEAddress.cpp (hunk 1 above)
Comment (Jeroen88) [reply src/NimBLEAddress.cpp]: There is always just one good reason for not using existing implementations: being unaware of it :D. I'll change it here and in NimBLEUUID too

File: src/NimBLEAddress.cpp (hunk 2 above)
Comment (Jeroen88) [reply src/NimBLEAddress.cpp:69]: A real improvement ;)

**Pull request merged**